test/sql/copy/parquet/writer/parquet_write_field_id.test

```diff
@@ -0,0 +1,268 @@
+----
+Binder Error
+
+# can't have duplicate field id values either
+statement error
+copy (select range as i, range as j from range(10)) to '__TEST_DIR__/my.parquet' (FIELD_IDS {i:42,j:42})
+----
+Binder Error
+
+# we don't have to supply a field_id for all columns
+statement ok
+copy (select range as i, range as j from range(10)) to '__TEST_DIR__/my.parquet' (FIELD_IDS {i:42})
+
+query I
+select field_id from parquet_schema('__TEST_DIR__/my.parquet') where name = 'i'
+----
+42
+
+query I
+select field_id from parquet_schema('__TEST_DIR__/my.parquet') where name = 'j'
+----
+NULL
+
+# but we can
+statement ok
+copy (select range as i, range as j from range(10)) to '__TEST_DIR__/my.parquet' (FIELD_IDS {i:42,j:43})
+
+query I
+select field_id from parquet_schema('__TEST_DIR__/my.parquet') where name = 'i'
+----
+42
+
+query I
+select field_id from parquet_schema('__TEST_DIR__/my.parquet') where name = 'j'
+----
+43
+
+# we can also specify the col like this
+statement ok
+copy (select range as i, range as j from range(10)) to '__TEST_DIR__/my.parquet' (FIELD_IDS {i:{field_id:42},j:{field_id:43}})
+
+query I
+select field_id from parquet_schema('__TEST_DIR__/my.parquet') where name = 'i'
+----
+42
+
+query I
+select field_id from parquet_schema('__TEST_DIR__/my.parquet') where name = 'j'
+----
+43
+
+# i is not a nested type, so we can't specify nested field ids
+statement error
+copy (select range as i, range as j from range(10)) to '__TEST_DIR__/my.parquet' (FIELD_IDS {i:{field_id:42,nested:{j:43}}})
+----
+Binder Error
+
+# we tested a non-nested column type, now do all the nested types so we test all the code paths
+
+# list
+statement ok
+copy (select range(range, range + 3) as i from range(10)) to '__TEST_DIR__/my.parquet' (FIELD_IDS {i:{field_id:42,nested:{element:43}}})
+
+query I
+select field_id from parquet_schema('__TEST_DIR__/my.parquet') where name = 'i'
+----
+42
+
+query I
+select field_id from parquet_schema('__TEST_DIR__/my.parquet') where name = 'element'
+----
+43
+
+# we don't have to specify a field_id for the top-level list, we can also just specify for the nested children
+statement ok
+copy (select range(range, range + 3) as i from range(10)) to '__TEST_DIR__/my.parquet' (FIELD_IDS {i:{nested:{element:43}}})
+
+query I
+select field_id from parquet_schema('__TEST_DIR__/my.parquet') where name = 'i'
+----
+NULL
+
+query I
+select field_id from parquet_schema('__TEST_DIR__/my.parquet') where name = 'element'
+----
+43
+
+# list child is always called "element"
+statement error
+copy (select range(range, range + 3) as i from range(10)) to '__TEST_DIR__/my.parquet' (FIELD_IDS {i:{field_id:42,nested:{elem:43}}})
+----
+Binder Error: Column name "elem" in FIELD_IDS not found. Available column names: [element]
+
+# struct
+statement ok
+copy (select {f : range} as i from range(10)) to '__TEST_DIR__/my.parquet' (FIELD_IDS {i:{field_id:42,nested:{f:43}}})
```

Comment:
Maybe inlining the children is nicer, e.g. `{i: {field_id: 42, f: 43}}`, or `__id` in case of name collisions.